### PR TITLE
Move e2e to GH actions and consolidate all tests in a single WF

### PR DESCRIPTION
**Pull request by benjamin-bergia**
.

### CODE REVIEW

File: backend/tests/unit/records/marshmallow/literature/common/test_external_system_identifier.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ class TestSchema(Schema):
                 "url_name": "OSTI Information Bridge Server",
             },
             {
-                "url_link": "https://zbmath.org/?q=an%3Azblatt-id",  # noqa
+                "url_link": "http://www.zentralblatt-math.org/zmath/en/search/?an=zblatt-id",  # noqa
```
Comment (pazembrz) [inline backend/tests/unit/records/marshmallow/literature/common/test_external_system_identifier.py:61]: Why this change in this PR?

Comment (michamos) [reply backend/tests/unit/records/marshmallow/literature/common/test_external_system_identifier.py:61]: looks like a wrong conflict resolution during rebase. I made the change to `zbmath.org` recently.

**Review by drjova — APPROVED** [review]

**Review by pazembrz — APPROVED** [review]
:+1:
